File: src/Backend.js
const express = require('express');
const bodyParser = require('body-parser');
const cors = require('cors'); // Import CORS middleware

const app = express();
const port = 3001;
app.use(cors()); // Enable CORS for all requests

app.use(bodyParser.json());

let notes = ["Note 1 from Node.js", "Note 2 from Node.js"];

app.get('/notes', (req, res) => {
  res.json({ notes });
});

app.post('/notes', (req, res) => {
  const { content } = req.body;
  notes.push(content);
  res.json({ message: 'Note added successfully' });
});

app.listen(port, () => {
  console.log(`Node.js backend listening at http://localhost:${port}`);
});

File: src/Backend.py
from flask import Flask, jsonify,request
from flask_cors import CORS
app = Flask(__name__)
CORS(app)
notes = ["Note 1 from Python", "Note 2 from Python"]

@app.route('/notes')
def get_notes():
    return jsonify({'notes': notes})

@app.route('/notes', methods=['POST'])
def add_note():
    content = request.json.get('content')
    notes.append(content)
    return jsonify({'message': 'Note added successfully'})

if __name__ == '__main__':
    app.run(port=5000)
